### ISSUE TITLE
Wrong sqlite version on iOS

### ISSUE THREAD

**Issue by lex-github** [simolus3/moor#889]
Hello, i am using moor:^3.3.1 in my flutter project. My code contains `open.overrideFor(OperatingSystem.iOS, () => DynamicLibrary.open(libraryNextToScript.path)); ` but `sqlite.version` outputs `2.28.0` instead of `3.32.3` which i see in `sqlite3.c` of sqlite3 pod.
Am i doing something wrong, can minimal example how to get latest sqlite version on iOS be provided?

**Comment by simolus3** [comment]
> `libraryNextToScript.path`

What does this path point to? If you're using the `sqlite3` pod, you should probably use `DynamicLibrary.process()` to open sqlite.

**Comment by lex-github** [comment]
Hello, thaks for reply and sorry for late response, i was actually using `DynamicLibrary.executable()` not `DynamicLibrary.open(libraryNextToScript.path)` and `3.28.00`. Tried `DynamicLibrary.process()` and got `3.28.00` version still.

**Comment by simolus3** [comment]
Sorry for the slow response. Unfortunately I don't know what could be causing this. We have an integration test that prints the sqlite3 version [over here](https://github.com/simolus3/sqlite3.dart/runs/1287717728?check_suite_focus=true), which reveals that we have 3.32.3 available. Do you use some kind of custom build setup on iOS that could be causing this?

**Comment by ab-rdarts** [comment]
same here. using moor and sqlite3_flutter_libs without moor_ffi. iOS11 simulator shows system version of sqlite 3.19.3 instead of 3.32.3.

**Comment by lex-github** [comment]
Still didn't have time to check example but i definitely will, thanks for link. Will compare with my setup and let you know.

**Comment by kuhnroyal** [comment]
See https://github.com/simolus3/moor/issues/940#issuecomment-735751647 for some new information.
The system provided versions can be found in this thread: https://stackoverflow.com/questions/14288128/what-version-of-sqlite-does-ios-provide
That list also confirms @ab-rdarts iOS11 - `3.19.3` combination
@lex-github `3.28.0` probably was an iOS13 device?

**Comment by simolus3** [comment]
Oh no, I expanded the test to also print compile-time flags used to compile sqlite3 and they don't match the podspec :scream: I'm sorry for all the confusion and trouble here.

I'll have to investigate why sqlite3 isn't actually included in the app if we use its podspec, maybe it helps if we also include `sqlite3.h` into the plugin. What's weird is that we open sqlite3 with `DynamicLibrary.process`: Is the system's sqlite3 just loaded into every process by default?? I don't understand how we're able to lookup sqlite3's symbols through that if we're not properly including sqlite3.

**Comment by kuhnroyal** [comment]
From my experience I would have to link sqlite3 in the XCode config to actually use it but the system library probably is always loaded/available.

**Comment by kuhnroyal** [comment]
I tried including the `sqlite3.h` header but that didn't change anything.

**Comment by kuhnroyal** [comment]
The included pod makes this a statically linked library. Not sure that can be used with `DynamicLibrary` at all. Maybe sqlite needs to be provided as `dylib`.

**Comment by simolus3** [comment]
I was under the impression that we could use statically linked code with `DynamicLibrary.process`. I'll take a look at making it a dylib.

**Comment by kuhnroyal** [comment]
> I was under the impression that we could use statically linked code with DynamicLibrary.process.
It says so here: https://flutter.dev/docs/development/platform-integration/c-interop
Maybe point it to the `.framework`?

Ok tried this real quick and it works!

```dart
return DynamicLibrary.open('sqlite3.framework/sqlite3');
```
The framework actually contains/is a dynamic library.

**Comment by kuhnroyal** [comment]
This is fixed in master for `sqlite3`/`sqlite3_flutter_libs` for non-custom sqlite libraries (meaning you use the one provided by sqlite3_flutter_libs).

Until it is released you can use this workaround:
```dart
import 'dart:ffi';
import 'package:sqlite3/open.dart';

// TODO Remove once sqlite3_flutter_libs is released
open.overrideFor(OperatingSystem.iOS, () => DynamicLibrary.open('sqlite3.framework/sqlite3'));
```

**Comment by simolus3** [comment]
I published Peter's fix in `sqlite3: 0.1.9-nullsafety.2`. Let me know if you need it on the stable channel too, I can release another update that doesn't need the beta SDK.
